app/src/main/java/com/davidtomas/taskyapp/features/agenda/data/photo/repository/PhotoRepositoryImpl.kt

```diff
@@ -6,6 +6,16 @@ import com.davidtomas.taskyapp.features.agenda.domain.repository.PhotoRepository
 class PhotoRepositoryImpl(
     private val photoLocalSource: PhotoLocalSource
 ) : PhotoRepository {
+
+    private val imageDataStore = mutableMapOf<String, ByteArray>()
+
+    override suspend fun saveImageData(key: String, data: ByteArray) {
+        imageDataStore[key] = data
+    }
+
+    override suspend fun getImageData(key: String): ByteArray? {
+        return imageDataStore[key]
+    }
```

Comment:
Saving all images in-memory doesn't seem very safe 😅 
Out of memory errors are waiting for you. And one process death and it's aaall gone